sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsClientTests.java

```diff
@@ -173,7 +176,7 @@ public void checkForRepeatabilityOptions(HttpClient httpClient) {
         client = setupSyncClient(builder, "checkForRepeatabilityOptions");
         // Action & Assert
         Response<Iterable<SmsSendResult>> response = client.sendWithResponse(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, null, Context.NONE);
-        String bodyRequest = new String(response.getRequest().getBody().blockLast().array());
+        String bodyRequest = StandardCharsets.UTF_8.decode(response.getRequest().getBody().blockLast()).toString();
```

Comment:
Any reason for this change?

Reply:
Some binary data contents serve read-only buffers.
```
java.nio.ReadOnlyBufferException
	at java.base/java.nio.ByteBuffer.array(ByteBuffer.java:1051)
	at com.azure.communication.sms@1.2.0-beta.1/com.azure.communication.sms.SmsClientTests.checkForRepeatabilityOptions(SmsClientTests.java:176)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

Reply:
Would a better option be `response.getRequest().getBodyAsBinaryData().toString()`?

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/28796